scio-hdfs/src/main/scala/com/spotify/scio/hdfs/package.scala

```diff
@@ -110,11 +110,28 @@ package object hdfs {
 
     /** Save this SCollection as a text file on HDFS. */
     // TODO: numShards
-    def saveAsHdfsTextFile(path: String, username: String = null): Future[Tap[String]] = {
+    def saveAsHdfsTextFile(path: String,
+                           username: String = null,
+                           conf: Configuration = null): Future[Tap[String]] = {
+      val _conf = Option(conf).getOrElse {
+        val newConf = new Configuration()
+        // since we are writing to HDFS - which we assume is remote to DF, let's compress the data.
+        // Deflate level can be between [1-9], the higher the better compression but more time
+        // spend. It seems that after 6 - there there is no big gain in size.
+        newConf.setBoolean(FileOutputFormat.COMPRESS, true)
+        newConf.set(FileOutputFormat.COMPRESS_CODEC, classOf[DefaultCodec].getName)
+        newConf.set(FileOutputFormat.COMPRESS_TYPE, "BLOCK")
+        newConf.setInt(AvroOutputFormat.DEFLATE_LEVEL_KEY, 6)
+        newConf
+      }
+
       val sink = if (username != null) {
-        new SimpleAuthHadoopFileSink(path, classOf[TextOutputFormat[NullWritable, Text]], username)
+        new SimpleAuthHadoopFileSink(path,
+                                     classOf[TextOutputFormat[NullWritable, Text]],
```

Comment:
maybe extract `classOf[TextOutputFormat[NullWritable, Text]]` in both if/else into a `val` and this will fit on 1 line.